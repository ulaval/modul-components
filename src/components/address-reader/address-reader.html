<address class="m-address-reader">
    <template v-if="subBuilding !== ''">{{ subBuilding }} -</template> {{ buildingNumber }} {{ street }} <br />
    {{ city }}, {{ province }} <br />
    {{ country }} <br />
    {{ postalCode }}
</address>
