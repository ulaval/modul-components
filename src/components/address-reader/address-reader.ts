import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import Address from '../../utils/address-lookup/address';
import { ModulVue } from '../../utils/vue/vue';
import WithRender from './address-reader.html';

const BUILDING_NUMBER: string = 'buildingNumber';
const STREET: string = 'street';
const CITY: string = 'city';
const POSTAL_CODE: string = 'postalCode';
const SUB_BUILDING: string = 'subBuilding';
const COUNTRY: string = 'country';
const PROVINCE: string = 'province';

export interface DisplayableAddress {
    buildingNumber: string;
    street: string;
    city: string;
    postalCode: string;
    subBuilding: string;
    country: string;
    province: string;
}

export enum ProvinceKey {
    PROVINCE = 'province',
    PROVINCE_CODE = 'provinceCode'
}

export enum CountryKey {
    COUNTRY = 'country',
    COUNTRY_ISO2 = 'countryIso2',
    COUNTRY_ISO3 = 'countryIso3'
}
@WithRender
@Component
export class MAddressReader extends ModulVue {
    @Prop({
        required: true,
        validator: (value: any) => value !== undefined
    })
    address: Address;

    @Prop({ default: () => { } })
    filters: { [field: string]: (value: string) => string };

    @Prop({
        default: CountryKey.COUNTRY,
        validator: (value: string) => value === CountryKey.COUNTRY
            || value === CountryKey.COUNTRY_ISO2
            || value === CountryKey.COUNTRY_ISO3
    })
    countryKey: string;

    @Prop({
        default: ProvinceKey.PROVINCE,
        validator: (value: string) => value === ProvinceKey.PROVINCE
            || value === ProvinceKey.PROVINCE_CODE
    })
    provinceKey: string;

    get buildingNumber(): string {
        return this.filterString(BUILDING_NUMBER);
    }

    get street(): string {
        return this.filterString(STREET);
    }

    get city(): string {
        return this.filterString(CITY);
    }

    get postalCode(): string {
        return this.filterString(POSTAL_CODE);
    }

    get subBuilding(): string {
        return this.filterString(SUB_BUILDING);
    }

    get country(): string {
        return this.filterString(COUNTRY, this.countryKey);
    }

    get province(): string {
        return this.filterString(PROVINCE, this.provinceKey);
    }

    get filteredAddress(): DisplayableAddress {
        return {
            buildingNumber: this.buildingNumber,
            street: this.street,
            city: this.city,
            postalCode: this.postalCode,
            subBuilding: this.subBuilding,
            country: this.country,
            province: this.province
        };
    }

    private filterString(key: string, subKey: string = ''): string {
        let value: string = '';
        if (subKey !== '') {
            value = this.address[key][subKey];
        } else {
            value = this.address[key];
        }
        return (this.filters[key]) ? this.filters[key](value) : value;
    }
}
