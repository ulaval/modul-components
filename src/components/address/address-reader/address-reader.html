<address class="m-address-reader">
    <slot :filtered-address="filteredAddress">
        <template v-if="subBuilding !== ''">{{ subBuilding }} -</template> {{ buildingNumber }} {{ street }} <br />
        {{ city }} <span v-if="hasProvince">, {{ province }}</span><br />
        {{ country }} <br />
        {{ postalCode }}
    </slot>
</address>
