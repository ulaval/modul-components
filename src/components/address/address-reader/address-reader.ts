import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import Address, { AddressField, CountryKey, ProvinceKey } from '../../../utils/address-lookup/address';
import { ModulVue } from '../../../utils/vue/vue';
import WithRender from './address-reader.html';

export interface DisplayableAddress {
    buildingNumber: string;
    street: string;
    city: string;
    postalCode: string;
    subBuilding: string;
    country: string;
    province: string;
}

export type addressFieldFilter = (value: string) => string;
export type addressesFilters = { [field: string]: addressFieldFilter };

export interface AddressReaderProps {
    address: Address;
    filters: { [field: string]: addressFieldFilter };
    countryKey: CountryKey;
    provinceKey: ProvinceKey;
}

@WithRender
@Component
export class MAddressReader extends ModulVue implements AddressReaderProps {
    @Prop({
        required: true,
        validator: (value: any) => value !== undefined
    })
    address: Address;

    @Prop({ default: () => ({}) })
    filters: addressesFilters;

    @Prop({
        default: CountryKey.COUNTRY,
        validator: (value: string) => value === CountryKey.COUNTRY
            || value === CountryKey.COUNTRY_ISO2
            || value === CountryKey.COUNTRY_ISO3
    })
    countryKey: CountryKey;

    @Prop({
        default: ProvinceKey.PROVINCE,
        validator: (value: string) => value === ProvinceKey.PROVINCE
            || value === ProvinceKey.PROVINCE_CODE
    })
    provinceKey: ProvinceKey;

    get buildingNumber(): string {
        return this.filterString(AddressField.BUILDING_NUMBER);
    }

    get street(): string {
        return this.filterString(AddressField.STREET);
    }

    get city(): string {
        return this.filterString(AddressField.CITY);
    }

    get postalCode(): string {
        return this.filterString(AddressField.POSTAL_CODE);
    }

    get subBuilding(): string {
        return this.filterString(AddressField.SUB_BUILDING);
    }

    get country(): string {
        return this.filterString(AddressField.COUNTRY, this.countryKey);
    }

    get province(): string {
        return this.filterString(AddressField.PROVINCE, this.provinceKey);
    }

    get hasProvince(): boolean {
        return this.filterString(AddressField.PROVINCE, this.provinceKey).length > 0;
    }

    get filteredAddress(): DisplayableAddress {
        return {
            buildingNumber: this.buildingNumber,
            street: this.street,
            city: this.city,
            postalCode: this.postalCode,
            subBuilding: this.subBuilding,
            country: this.country,
            province: this.province
        };
    }

    private filterString(key: string, subKey: string = ''): string {
        let value: string = '';
        if (subKey !== '' && this.address[key] && this.address[key][subKey]) {
            value = this.address[key][subKey];
        } else if (subKey === '' && this.address[key]) {
            value = this.address[key];
        }
        return (this.filters[key] && value.length > 0) ? this.filters[key](value) : value;
    }
}
