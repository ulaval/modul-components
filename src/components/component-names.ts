export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
