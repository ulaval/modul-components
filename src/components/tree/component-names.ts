export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
