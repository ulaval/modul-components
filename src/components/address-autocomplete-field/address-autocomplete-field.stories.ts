import { withA11y } from '@storybook/addon-a11y';
import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { CountryKey, ProvinceKey } from '../address-reader/address-reader';
import { ADDRESS_AUTOCOMPLETE_FIELD_NAME } from '../component-names';
import MAddressAutocompleteField from './address-autocomplete-field';


storiesOf(`${componentsHierarchyRootSeparator}${ADDRESS_AUTOCOMPLETE_FIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('default', () => ({
        components: { MAddressAutocompleteField },
        data: () => ({
            address: {},
            props: {
                language: 'fr',
                origin: 'canada',
                filters: {
                    'street': (value: string) => value.charAt(0).toLowerCase() + value.slice(1)
                },
                countryKey: CountryKey.COUNTRY,
                provinceKey: ProvinceKey.PROVINCE_CODE
            }
        }),
        methods: {
            clear(): void {
                (this as any).address = {};
            }
        },
        template: `
        <div>
            <div>
                <${ADDRESS_AUTOCOMPLETE_FIELD_NAME} v-model="address" v-bind="props">
                </${ADDRESS_AUTOCOMPLETE_FIELD_NAME}>
            </div>
            <div style="border-top: 1px solid #000; margin-top: 50px;">Address from autocomplete field : {{ address }}</div>
            <div><a @click.prevent="clear">clear result</a></div>
        </div>`
    }));
