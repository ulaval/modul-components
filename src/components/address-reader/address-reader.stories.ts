import { withA11y } from '@storybook/addon-a11y';
import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { ADDRESS_READER } from '../component-names';
import { MAddressReader } from './address-reader';

storiesOf(`${componentsHierarchyRootSeparator}${ADDRESS_READER}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('default', () => ({
        components: { MAddressReader },
        data: () => ({
            address: {
                language: 'FRE',
                alternativeLanguages: ['FRE', 'ENG'],
                buildingNumber: '2325',
                city: 'Québec',
                country: {
                    country: 'Canada',
                    countryIso2: 'CA',
                    countryIso3: 'CAN'
                },
                province: {
                    province: 'Québec',
                    provinceCode: 'QC'
                },
                street: `Rue de l'Université`,
                postalCode: 'G1V 0A6',
                subBuilding: '',
                label: `2325 Rue de l'Université`
            },
            countryKey: 'country',
            provinceKey: 'provinceCode',
            filters: {
                street: (value: string): string => value.charAt(0).toLowerCase() + value.slice(1)
            }
        }),
        template: `<div>
            <${ADDRESS_READER}
                :address="address"
                :country-key="countryKey"
                :province-key="provinceKey"
                :filters="filters">
            </${ADDRESS_READER}>
        </div>`
    }))
    .add('Restructuring using scoped-slot', () => ({
        components: { MAddressReader },
        data: () => ({
            address: {
                language: 'FRE',
                alternativeLanguages: ['FRE', 'ENG'],
                buildingNumber: '2325',
                city: 'Québec',
                country: {
                    country: 'Canada',
                    countryIso2: 'CA',
                    countryIso3: 'CAN'
                },
                province: {
                    province: 'Québec',
                    provinceCode: 'QC'
                },
                street: `Rue de l'Université`,
                postalCode: 'G1V 0A6',
                subBuilding: '',
                label: `2325 Rue de l'Université`
            },
            countryKey: 'countryIso2',
            provinceKey: 'province',
            filters: {
                street: (value: string): string => value.charAt(0).toLowerCase() + value.slice(1)
            }
        }),
        template: `<div>
            <${ADDRESS_READER}
                :address="address"
                :country-key="countryKey"
                :province-key="provinceKey"
                :filters="filters"
                v-slot:default="{ filteredAddress }">
                {{ filteredAddress.postalCode }} <br />
                {{ filteredAddress.country }} <br />
                {{ filteredAddress.city }}, {{ filteredAddress.province }} <br />
                {{ filteredAddress.buildingNumber }} {{ filteredAddress.street }} <template v-if="filteredAddress.subBuilding !== ''">Appartement / bureau : {{ filteredAddress.subBuilding }} </template>
            </${ADDRESS_READER}>
        </div>`
    }));
