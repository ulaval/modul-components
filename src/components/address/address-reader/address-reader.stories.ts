import { storiesOf } from '@storybook/vue';
import { componentsHierarchyRootSeparator } from '../../../../conf/storybook/utils';
import { CountryKey, ProvinceKey } from '../../../utils/address-lookup/address';
import { ADDRESS_READER } from '../../component-names';
import { MAddressReader } from './address-reader';

storiesOf(`${componentsHierarchyRootSeparator}/address/${ADDRESS_READER}`, module)
    .add('default', () => ({
        components: { MAddressReader },
        data: () => ({
            address: {
                buildingNumber: '2325',
                city: 'Québec',
                country: {
                    country: 'Canada',
                    countryIso2: 'CA'
                },
                province: {
                    province: 'Québec',
                    provinceCode: 'QC'
                },
                street: `Rue de l'Université`,
                postalCode: 'G1V 0A6',
                subBuilding: ''
            },
            countryKey: CountryKey.COUNTRY,
            provinceKey: ProvinceKey.PROVINCE_CODE,
            filters: {
                street: (value: string): string => value.charAt(0).toLowerCase() + value.slice(1)
            }
        }),
        template: `<div>
            <${ADDRESS_READER}
                :address="address"
                :country-key="countryKey"
                :province-key="provinceKey"
                :filters="filters">
            </${ADDRESS_READER}>
        </div>`
    }))
    .add('Restructuring using scoped-slot', () => ({
        components: { MAddressReader },
        data: () => ({
            address: {
                buildingNumber: '2325',
                city: 'Québec',
                country: {
                    country: 'Canada',
                    countryIso2: 'CA',
                    countryIso3: 'CAN'
                },
                province: {
                    province: 'Québec',
                    provinceCode: 'QC'
                },
                street: `Rue de l'Université`,
                postalCode: 'G1V 0A6',
                subBuilding: ''
            },
            countryKey: CountryKey.COUNTRY,
            provinceKey: ProvinceKey.PROVINCE_CODE,
            filters: {
                street: (value: string): string => (value !== '') ? value.charAt(0).toLowerCase() + value.slice(1) : ''
            }
        }),
        template: `<div>
            <${ADDRESS_READER}
                :address="address"
                :country-key="countryKey"
                :province-key="provinceKey"
                :filters="filters"
                v-slot:default="{ filteredAddress : address }">
                {{ address.postalCode }} <br />
                {{ address.country }} <br />
                {{ address.city }}, {{ address.province }} <br />
                {{ address.buildingNumber }} {{ address.street }} <template v-if="address.subBuilding !== ''">Appartement / bureau : {{ address.subBuilding }} </template>
            </${ADDRESS_READER}>
        </div>`
    }))
    .add('empty province', () => ({
        components: { MAddressReader },
        data: () => ({
            address: {
                buildingNumber: '2325',
                city: 'Québec',
                country: {
                    country: 'Canada',
                    countryIso2: 'CA'
                },
                province: {
                    province: '',
                    provinceCode: ''
                },
                street: `Rue de l'Université`,
                postalCode: 'G1V 0A6',
                subBuilding: ''
            },
            countryKey: CountryKey.COUNTRY,
            provinceKey: ProvinceKey.PROVINCE_CODE,
            filters: {
                street: (value: string): string => value.charAt(0).toLowerCase() + value.slice(1)
            }
        }),
        template: `<div>
            <${ADDRESS_READER}
                :address="address"
                :country-key="countryKey"
                :province-key="provinceKey"
                :filters="filters">
            </${ADDRESS_READER}>
        </div>`
    }));
