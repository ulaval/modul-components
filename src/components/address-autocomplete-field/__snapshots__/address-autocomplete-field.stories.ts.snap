// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|/address/m-address-autocomplete-field default 1`] = `
<div>
  <div>
    <div
      class="m-address-autocomplete-field"
    >
      <div
        class="m-address-lookup-field"
      >
        <m-autocomplete
          label="Rechercher une adresse"
          placeholder="Entrez une adresse Ã  rechercher"
          results=""
        />
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    style="border-top: 1px solid #000; margin-top: 50px;"
  >
    Address from autocomplete field : {}
  </div>
   
  <div>
    <a>
      clear result
    </a>
  </div>
</div>
`;
